wagtailio/navigation/templatetags/navigation_tags.py

```diff
@@ -0,0 +1,14 @@
+from django import template
+
+register = template.Library()
+
+
+@register.inclusion_tag("patterns/includes/menu_footer.html", takes_context=True)
+def footer_navigation(context):
+    request = context["request"]
+    return {
+        "footer_navigation": context["settings"]["navigation"][
```

Comment:
Perhaps this could be addressed by following the approach used here?
 https://github.com/wagtail/wagtail.org/blob/20db37b74dd38a9e9082277289534b6909275a56/wagtailio/project_styleguide/pattern_contexts.py#L8-L10

Reply:
I am not fully enamored by that approach as it makes two queries.

that bit can be 
```python
if nav := = NavigationSettings.objects.first():
  context["settings"] = {"navigation": {"NavigationSettings": nav}} 
```

note `:=` is the [walrus operator](https://realpython.com/python-walrus-operator/)

---

For this, it is best to wrap it in a try/except:

```python
try:
   footer_navigation = context["settings"]["navigation"]["NavigationSettings"].footer_navigation
except (KeyError, AttributeError):
   footer_navigation = None

return {
   "footer_navigation": footer_navigation,
   "request": request
}
```

question: do you need the request in the template context?

Reply:
> question: do you need the request in the template context?

You're right, I don't think it's needed! I've removed it in 4b347d68

Reply:
> For this, it is best to wrap it in a try/except:

Thanks, done in 4b347d6